### PR TITLE
Using constructor property promotion in the SecurityBundle

### PR DESCRIPTION
**Pull request by mamazu**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | #7435 #7325 
| License | MIT
| Documentation PR | -

#### What's in this PR?
Using constructor property promotion in the SecurityBundle

#### Why?
There isn't much properties to promote here so we do type fixing. :D

### CODE REVIEW

File: src/Sulu/Bundle/SecurityBundle/Security/Exception/TokenAlreadyRequestedException.php [hunk 1]
```diff
@@ -19,16 +19,11 @@
...
     /**
-     * The time interval in which only one token can be requested.
-     *
-     * @var \DateInterval
+     * @param \DateInterval $interval the time interval in which only one token can be requested
      */
-    private $interval;
-
-    public function __construct($interval)
+    public function __construct(private \DateInterval $interval)
```
Comment (mamazu) [inline src/Sulu/Bundle/SecurityBundle/Security/Exception/TokenAlreadyRequestedException.php:24]: Hoisted from `toArray`

File: src/Sulu/Bundle/SecurityBundle/EventListener/UserLocaleListener.php [hunk 2]
```diff
@@ -17,36 +17,16 @@
...
-                LocaleAwareInterface::class,
-                \get_class($translator)
-            ));
-        }
-
-        $this->tokenStorage = $tokenStorage;
-        $this->translator = $translator;
+    public function __construct(
+        private TokenStorageInterface $tokenStorage,
+        private LocaleAwareInterface $translator
```
Comment (mamazu) [inline src/Sulu/Bundle/SecurityBundle/EventListener/UserLocaleListener.php:28]: Using the type system for type checks.

File: src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php [hunk 3]
```diff
@@ -35,69 +35,23 @@ class CreateUserCommand extends Command
...
+        private EntityManagerInterface $entityManager,
+        private UserRepository $userRepository,
+        private RoleRepositoryInterface $roleRepository,
+        private ContactRepositoryInterface $contactRepository,
+        private LocalizationManagerInterface $localizationManager,
+        private SaltGenerator $saltGenerator,
+        private $passwordHasherFactory,
+        /**
+         * @var string[]
+         */
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php]: Would move the phpdoc up to the `@param` like we already do for the encoder

**Review by alexander-schranz — APPROVED** [review]

File: src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php (hunk 3 above)
Comment (mamazu) [reply src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php]: Will do.

**Comment by mamazu** [comment]
Or you do it. :D

**Comment by alexander-schranz** [comment]
@mamazu 😆

**Comment by alexander-schranz** [comment]
Hate that Github don't allow todo suggestions on lines which are not part of the diff.

**Pull request merged**